plugins/core/Standard/qCSF/include/qCSFCommands.h

```diff
@@ -0,0 +1,270 @@
+//#                                                                                     #
+//#                                     Copyright ©                                     #
+//#               RAMM laboratory, School of Geography, Beijing Normal University       #
+//#                               (http://ramm.bnu.edu.cn/)                             #
+//#                                                                                     #
+//#                      Wuming Zhang; Jianbo Qi; Peng Wan; Hongtao Wang                #
+//#                                                                                     #
+//#                      contact us: 2009zwm@gmail.com; wpqjbzwm@126.com                #
+//#                                                                                     #
+//#######################################################################################
+
+// A mex version for programming in Matlab is at File Exchange of Mathworks website:
+// http://www.mathworks.com/matlabcentral/fileexchange/58139-csf--ground-filtering-of-point-cloud-based-on-cloth-simulation
+
+#ifndef QCSF_PLUGIN_COMMANDS_HEADER
+#define QCSF_PLUGIN_COMMANDS_HEADER
+
+//CloudCompare
+#include "ccCommandLineInterface.h"
+
+//Local
+#include "ccCSFDlg.h"
+#include "CSF.h"
+
+static const char COMMAND_CSF[] = "CSF"; 
+static const char COMMAND_CSF_SCENE[] = "SCENES";
+static const char COMMAND_CSF_SCENE_SLOPE[] = "SLOPE";
+static const char COMMAND_CSF_SCENE_RELIEF[] = "RELIEF";
+static const char COMMAND_CSF_SCENE_FLAT[] = "FLAT";
+static const char COMMAND_CSF_PROC_SLOPE[] = "PROC_SLOPE"; 
+static const char COMMAND_CSF_CLOTH_RESOLUTION[] = "CLOTH_RESOLUTION";
+static const char COMMAND_CSF_MAX_ITERATION[] = "MAX_ITERATION";
+static const char COMMAND_CSF_CLASS_THRESHOLD[] = "CLASS_THRESHOLD";
+static const char COMMAND_CSF_EXPORT_GROUND[] = "EXPORT_GROUND";
+static const char COMMAND_CSF_EXPORT_OFFGROUND[] = "EXPORT_OFFGROUND";
+
+
+struct CommandCSF : public ccCommandLineInterface::Command
+{
+	CommandCSF() : ccCommandLineInterface::Command("CSF", COMMAND_CSF) {}
+
+	virtual bool process(ccCommandLineInterface& cmd) override
+	{
+		cmd.print("[CSF]");
+
+		if (cmd.clouds().size() < 1) {
+			cmd.error("No cloud loaded (1 is expected)");
+			return false;
+		}
+
+		ccPointCloud* pc = cmd.clouds()[0].pc;
+
+		//Convert CC point cloud to CSF type
+		unsigned count = pc->size();
+		wl::PointCloud csfPC;
+		try {
+			csfPC.reserve(count);
+		}
+		catch (const std::bad_alloc&) {
+			return cmd.error("Not enough memory!");
+		}
+
+		for (unsigned i = 0; i < count; i++) {
+			const CCVector3* P = pc->getPoint(i);
+			wl::Point tmpPoint;
+			//tmpPoint.x = P->x;
+			//tmpPoint.y = P->y;
+			//tmpPoint.z = P->z;
+			tmpPoint.x = P->x;
+			tmpPoint.y = -P->z;
+			tmpPoint.z = P->y;
+			csfPC.push_back(tmpPoint);
+		}
+
+		//initial parameters
+		static bool csf_postprocessing = false;
```

Comment:
why static?

Reply:
In any case, in this project we set a upper case letters at the beginning of static variable names, and lower case for the others.